articles/mvc-mvp-mvvm/index.md

```diff
@@ -0,0 +1,52 @@
+
+#### Model View View-Model
+
+In [MVVM](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel), the view is independent of the model. This means that the UI part is separated from the core logic of the application. It's the job of the view model to retrieve the data from the model and present them in the view. Also, the view model carries out the computational work of the view by handling the user events and updating the UI. This is because in MVVM the view is responsible for notifying the view-model about all the user action. As a result, we can assume that View and View-Model has a two-way data binding relationship.
+
+### Mind Map
+
+Since the above architectural designs are based around the idea of collaborating with their own structure to achieve a positive outcome, each component has its own set of responsibilities. Model and View are shared by all of the patterns mentioned above. The third aspect is what distinguishes them from one another. As a result, here is a simplified mind map to help you visualize what each pattern represents.
+
+![Architectural Patterns](/engineering-education/mvc-mvp-mvvm/img.jpg)
+
+### Summary
+
+So, in this article, we went through three architecture patterns that can be used while designing Android apps. While using design patterns in your apps is not mandatory, implementing them will definitely make the UI interface more user-friendly and will benefit in managing various functions and classes. This contributes to giving the application a modular structure. Often, implementing patterns can be time-consuming because each component must be built first and then unit tested before being released completely into the source code. Although the developer team must be meticulous in their future supervision and management because a single flaw or bug in the pattern may be harmful to the coherence of the architecture. Even so, one thing is certain: the code becomes less messy and more readable and it also performs efficiently in A/B testing. No one pattern is considered the best of the three; it essentially depends on the users' demand, developers' decision, and the needs of the application.
+

```

Comment:
Please include a summary where you compare all three architectural patterns. this improves the quality of the article.